improve profile preferences window

atari2600tim opened the pull request:
<!-- Keep the title short & concise so anyone non-technical can understand it,
     the title appears in PTB changelogs -->
#### Brief overview of PR changes/additions
It has changes on the preferences window, main thing is moving Accessibility to a new tab.

While I'm at it:
- dim the Clear stored media group when no profile open
- add a tab stop for that same Clear button
- dim the accessibility labels when no profile instead of just the drop boxes
- renamed `groupBox` to `groupBox_accessibility`
- moved `Show Spaces/Tabs` and `Show Line/Paragraphs` from Main Display tab to Editor tab
` moved `2D Room Symbol scaling factor` (visible only when map is opened) from Special Options to Mapper

#### Motivation for adding to Mudlet
The Save button was pushed off the screen on my system because the Special Options tab is so tall and the window is stretched to hold it.
#### Other info (issues closed, discussion etc)

atari2600tim committed:
initial

vadi2 commented:
What do you think of setting the tabs to be vertical instead of horizontal? At this point we have enough to necessitate that

atari2600tim commented:
It does seem pretty far to go.  I gave scrollwheel two spins to reach the end, other time clicked the right arrow a long time, and other time pressed the right arrow key a bunch of times.

vadi2 commented:
Works well for me. Could you ask some folks in #accessibility to test it too?

ironcross32 commented:
I see no issues from an accessibility point of view.

atari2600tim commented:
I tried it with NVDA and used keyboard to navigate just now.  If you open preferences with no profiles open then this ends up being an entire tab that has everything disabled.  I'm not sure if there should be some kind of hint text that could be attached to explain why it sounds empty.  Most of the other tabs are similar situation though, so this isn't any worse than it's been.

I noticed some other things aren't dimmed when no profile, so I'll add those in a bit.

ironcross32 commented:
Hmm, I normally don't open preferences with no profiles open, let me try that and see what it's like.

ironcross32 commented:
> Hmm, I normally don't open preferences with no profiles open, let me try that and see what it's like.

Yeah, I see nothing wrong here. It reads the tab, the arrows for scrolling the tab list, and the save button then wraps back around.

ironcross32 commented:
@atari2600tim If your NVDA is reading disabled text in this tab, ensure you're up to date. You should be on  version 2023.1
1. Press INSERT+N, Help > About NVDA
2. the version is listed there
3. Alternatively, INSERT+N > help > check for updates, download, say no to donations, let it run and try again.

atari2600tim commented:
My NVDA I think is acting properly and not reading the disabled text.  I just wanted to verify that tabs, arrows, save button, and cycle around is the right and expected behavior.  I've only used it with Mudlet, so I have no idea if other programs might present an explanation of some sort.  Maybe an IRC client would say "there are no server properties available because you haven't selected a server".

ironcross32 commented:
How it is presented now as expected behavior. I would say that nothing else needs to be done in this regard.

atari2600tim commented:
> I noticed some other things aren't dimmed when no profile, so I'll add those in a bit.

I was looking at something I mistakenly thought was associated with current profile, so disregard.

atari2600tim committed:
rearranged some more

Moved Show Spaces/Tabs and Show Line/Paragraphs from Main Display tab to a new Display options box in the Editor tab and adjusted the ones to fill that gap
The Special Options tab has logic to add 2D Room Symbol scaling factor if you look at that tab when the mapper is open.  I moved that to add itself at bottom of Mapper tab instead at bottom of Map view box.

atari2600tim commented:
Moved `Show Spaces/Tabs` and `Show Line/Paragraphs` from Main Display tab to a new Display options box in the Editor tab and adjusted the ones to fill that gap
The Special Options tab has logic to add `2D Room Symbol scaling factor` if you look at that tab when the mapper is open.  I moved that to add itself at bottom of Mapper tab instead at bottom of Map view box.

atari2600tim committed:
Merge branch 'Mudlet:development' into timSaveButton

atari2600tim committed:
Merge branch 'development' into timSaveButton
